src/Controller/Api/Legacy/PostController.php

```diff
@@ -0,0 +1,453 @@
+            $comment = $form->getData();
+            $comment->setPost($post);
+
+            if (!$comment->getStatus()) {
+                $comment->setStatus($post->getCommentsDefaultStatus());
+            }
+
+            $this->commentManager->save($comment);
+            $this->mailer->sendCommentNotification($comment);
+
+            return $comment;
+        }
+
+        return $form;
+    }
+
+    /**
+     * Updates a comment.
+     *
+     * @ApiDoc(
+     *  requirements={
+     *      {"name"="postId", "dataType"="string", "description"="Post identifier"},
+     *      {"name"="commentId", "dataType"="string", "description"="Comment identifier"}
+     *  },
+     *  input={"class"="sonata_news_api_form_comment", "name"="", "groups"={"sonata_api_write"}},
+     *  output={"class"="Sonata\NewsBundle\Model\Comment", "groups"={"sonata_api_read"}},
+     *  statusCodes={
+     *      200="Returned when successful",
+     *      400="Returned when an error has occurred while comment update",
+     *      404="Returned when unable to find comment"
+     *  }
+     * )
+     *
+     * @REST\View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
+     *
+     * @param string  $postId    Post identifier
+     * @param string  $commentId Comment identifier
+     * @param Request $request   Symfony request
+     *
+     * @throws NotFoundHttpException
+     * @throws HttpException
+     *
+     * @return Comment
+     */
+    public function putPostCommentsAction($postId, $commentId, Request $request)
+    {
+        $post = $this->getPost($postId);
+
+        if (!$post->isCommentable()) {
+            throw new HttpException(403, sprintf('Post (%s) not commentable', $postId));
+        }
+
+        $comment = $this->commentManager->find($commentId);
+
+        if (null === $comment) {
+            throw new NotFoundHttpException(sprintf('Comment (%s) not found', $commentId));
+        }
+
+        $comment->setPost($post);
+
+        $form = $this->formFactory->createNamed(null, 'sonata_news_api_form_comment', $comment, [
+            'csrf_protection' => false,
+        ]);
+
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $comment = $form->getData();
+            $this->commentManager->save($comment);
+
+            return $comment;
+        }
+
+        return $form;
+    }
+
+    /**
+     * Filters criteria from $paramFetcher to be compatible with the Pager criteria.
+     *
+     * @return array The filtered criteria
+     */
+    protected function filterCriteria(ParamFetcherInterface $paramFetcher)
+    {
+        $criteria = $paramFetcher->all();
+
+        unset($criteria['page'], $criteria['count']);
+
+        $criteria = array_filter($criteria, static function ($value): bool {
+            return null !== $value;
+        });
+
+        if (\array_key_exists('dateValue', $criteria)) {
+            $date = new \DateTime($criteria['dateValue']);
```

Comment:
Maybe we could use `\DateTimeImmutable` here.

Reply:
I'm not sure. IMO we should not do deep refactoring.